src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/Language/CSharpIdentifier.cs

```diff
@@ -59,9 +59,25 @@ public static void AppendSanitized(StringBuilder builder, ReadOnlySpan<char> inp
             builder.SetCapacityIfLarger(builder.Length + inputName.Length);
         }
 
-        foreach (var ch in inputName)
+        for (int i = 0; i < inputName.Length; i++)
```

Comment:
Reading through the issue for this, it seems the most important part if this change is not so much exactly what gets emitted, but rather that it matches the new project template, and template engine, exactly. A comment to that effect is probably worth adding.